src/plugins/discover/server/ui_settings.ts

```diff
@@ -38,6 +39,17 @@ export const uiSettings: Record<string, UiSettingsParams> = {
     category: ['discover'],
     schema: schema.arrayOf(schema.string()),
   },
+  [MAX_DOC_FIELDS_DISPLAYED]: {
+    name: i18n.translate('discover.advancedSettings.maxDocFieldsDisplayedTitle', {
+      defaultMessage: 'Maximum document column entries',
+    }),
+    value: 200,
+    description: i18n.translate('discover.advancedSettings.maxDocFieldsDisplayedText', {
+      defaultMessage: 'Maximum number of entries rendered in the document column',
+    }),
+    category: ['discover'],
+    schema: schema.number(),
+  },
```

Comment:
I'd suggest both title and description to be aligned with the new wording
```suggestion
  [MAX_DOC_FIELDS_DISPLAYED]: {
    name: i18n.translate('discover.advancedSettings.maxDocFieldsDisplayedTitle', {
      defaultMessage: 'Maximum document fields displayed',
    }),
    value: 200,
    description: i18n.translate('discover.advancedSettings.maxDocFieldsDisplayedText', {
      defaultMessage: 'Maximum number of fields rendered in the document column',
    }),
    category: ['discover'],
    schema: schema.number(),
  },
```